package.json

```diff
@@ -168,29 +177,30 @@
     "sinon": "^16.1.3",
     "stream-browserify": "^3.0.0",
     "string-replace-loader": "^3.1.0",
+    "terser-webpack-plugin": "^5.3.10",
     "tsx": "^4.7.0",
     "typescript": "^5.3.3",
     "unist-util-visit": "^2.0.3",
     "util": "^0.12.5",
     "webpack": "^5.90.3",
+    "webpack-bundle-analyzer": "^4.10.1",
     "webpack-cli": "^5.1.4",
-    "yargs": "^17.7.2",
-    "yarn-deduplicate": "^6.0.2"
+    "yargs": "^17.7.2"
   },
   "setupFiles": [
     "<rootDir>/src/setupTests.js"
   ],
-  "workspaces": {
-    "packages": [
-      "packages/*",
-      "docs"
-    ],
-    "nohoist": [
-      "**/@mui/monorepo"
-    ]
-  },
   "resolutions": {
-    "**/react-is": "^18.2.0",
-    "**/@types/node": "^18.19.23"
+    "react-is": "^18.2.0",
+    "@types/node": "^18.19.23"
+  },
+  "packageManager": "pnpm@8.15.1",
+  "pnpm": {
+    "patchedDependencies": {
+      "date-fns-jalali@2.13.0-0": "patches/date-fns-jalali@2.13.0-0.patch",
```

Comment:
Is this intended? It won't propagate to users

Reply:
Yes. It's just a replacement for the patch that already existed.
We don't need it propagated to end users, it's just so that we can avoid duplicate identifier problem because in our repo both `date-fns` packages co-exist. 🙈 
<img width="1166" alt="Screenshot 2024-04-19 at 11 36 05" src="https://github.com/mui/mui-x/assets/4941090/be2e7299-4e07-4290-a2f0-48e5be047ee7">

Do you have better suggestions on how to avoid this problem? 🤔